Additional Source Geometry Added to phot_pipeline

kglidic opened the pull request:
I have added a circular annulus as an option for the source geometry. I have also tested this geometry out. Here is a basic example of how one could use this geometry:

[06-09-2022_tshirt_srcGeometry_Test.pdf](https://github.com/eas342/tshirt/files/8873451/06-09-2022_tshirt_srcGeometry_Test.pdf)

kglidic committed:
Additional Source Geometry Added to phot_pipeline

I have added a circular annulus as an option for the source geometry. I have also tested this geometry out. Here is a basic example of how one could use this geometry:

#Assignimg a object phot
phot = phot_pipeline.phot(paramFile=paramFile") #create a photometric object
alteredParam = deepcopy(phot.param) # copy the original parameter file
alteredParam['srcGeometry']='CircularAnnulus' #Change the source geometry
alteredParam['srcEnd'] = 30 #Add a parameter for the source start radius
alteredParam['srcStart'] = 12 #Add a parameter for the source end radius
alteredParam['bkgGeometry'] = 'CircularAnnulus' #Background Geometry can also be a circular annulus
alteredParam['backEnd'] = 60 #Outer background radius
alteredParam['backStart'] = 30 #Inner background radius 
alteredParam['bkgMethod'] = 'mean'
phot2 = phot_pipeline.phot(directParam=alteredParam) #create new photometric object
phot2.showStarChoices(showAps=True,showPlot=True, apColor='red', backColor='pink', figSize=(30,20)) #Plot the source and background subtraction area

eas342 commented:
This is great! Please help by adding some explanation for this here:
https://tshirt.readthedocs.io/en/latest/phot_pipeline/phot_pipeline_parameters.html#source-aperture-photometry-geometry
It should say how to use the `srcStart` and `srcEnd`parameters and that with this geometry `apRadius` will be ignored.
The code can be found here:
https://github.com/eas342/tshirt/blob/master/docs/phot_pipeline/phot_pipeline_parameters.rst

kglidic committed:
Updated some of the geometry documentation

Added some documentation on the CircularAnnulus source geometry. Also edited some of the existing documentation for what I assumed were small errors.

kglidic committed:
Merge branch 'eas342:master' into master
